python/lsst/daf/persistence/access.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+
+#
+# LSST Data Management System
+# Copyright 2016 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+import cPickle
+import collections
+import importlib
+import os
+
+from lsst.daf.persistence import LogicalLocation, Policy, butlerExceptions, StorageList, Persistence
+import lsst.pex.policy as pexPolicy
+from .safeFileIo import SafeFilename
+
+# How to document Storage class protocol? Virtual base class seems somewhat non-pythonic?
+class Storage(object):
+
+    @staticmethod
+    def getMapperClass(root):
+        raise butlerExceptions.NotImplemented("getMapperClass is not implemented by derived class")
```

Comment:
I'm also not sure that a Storage should really know about a Mapper, which I thought is a more Repository thing.  I think that would also allow you to move the parent searching out of here, where it's hard-coded to be in terms of `_parent` links, and into the Repository code that knows about parent Repository relationships and their priorities.